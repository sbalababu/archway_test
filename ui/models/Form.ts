export interface QuotaForm {
  quota: number;
}

export interface CoreMemoryForm {
  core: number;
  memory: number;
}

export interface MemberForm {
  username: string;
  roles: object;
}
