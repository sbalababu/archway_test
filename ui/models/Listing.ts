export interface Filters {
  filter: string;
  behaviors: string[];
  statuses: string[];
}
