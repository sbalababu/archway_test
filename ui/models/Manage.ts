import { ComplianceType } from '../constants';

export interface Question {
  question: string;
  requester: string;
  updated: Date;
  id?: number;
  complianceGroupId?: number;
}

export interface ComplianceContent {
  name: ComplianceType;
  id?: number;
  description: string;
  questions: Question[];
}

export interface Link {
  name: string;
  description: string;
  url: string;
  id?: number;
  customLinkGroupId?: number;
}

export interface LinksGroup {
  name: string;
  description: string;
  links: Link[];
  id?: number;
}
