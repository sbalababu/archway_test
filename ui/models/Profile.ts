export interface Profile {
  name: string;
  username: string;
  distinguished_name: string;
  permissions: {
    risk_management: boolean;
    platform_operations: boolean;
  };
}
