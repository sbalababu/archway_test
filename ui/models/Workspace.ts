export interface YarnApplication {
  id: string;
  name: string;
  start_time?: Date;
}

export interface ResourcePoolsInfo {
  loading: boolean;
  error?: string;
  data: PoolInfo[];
}

export interface PoolInfo {
  resource_pool: string;
  applications: YarnApplication[];
}

export interface NamespaceInfo {
  name: string;
  tables: HiveTable[];
}

export interface NamespaceInfoList {
  loading: boolean;
  error?: string;
  data: NamespaceInfo[];
}

export interface HiveTable {
  name: string;
}

export interface MemberRemoveStatus {
  loading?: boolean;
  success?: boolean;
  error?: boolean;
}

export interface Member {
  distinguished_name: string;
  email?: string;
  name: string;
  removeStatus?: MemberRemoveStatus;
  data: any;
  topics: any;
  applications: any;
}

export interface Application {
  id: number;
  name: string;
  consumer_group: string;
  group: SecurityGroup;
  type?: string;
  logo?: string;
  language?: string;
  repository?: string;
}

export interface TopicGrant {
  id: number;
  group: SecurityGroup;
  actions: string;
  topic_access?: Date;
}

export interface KafkaTopic {
  id: number;
  name: string;
  partitions: number;
  replication_factor: number;
  managing_role: TopicGrant;
  readonly_role: TopicGrant;
}

export interface ResourcePool {
  id: number;
  pool_name: string;
  max_cores: number;
  max_memory_in_gb: number;
}

export interface SecurityGroup {
  common_name: string;
  distinguished_name: string;
  sentry_role: string;
  group_created?: Date;
  role_created?: Date;
  role_associated?: Date;
  attributes?: string[][];
}

export interface Compliance {
  phi_data?: boolean;
  pci_data?: boolean;
  pii_data?: boolean;
}

export interface DatabaseGrant {
  group: SecurityGroup;
  location_access?: Date;
  database_access?: Date;
}

export interface HiveAllocation {
  id: number;
  name: string;
  location: string;
  size_in_gb: number;
  consumed_in_gb: number;
  managing_group: DatabaseGrant;
  readwrite_group?: DatabaseGrant;
  readonly_group?: DatabaseGrant;
}

export interface ApprovalStatus {
  loading?: boolean;
  success?: boolean;
  error?: string;
}

export interface ApprovalItem {
  approver?: string;
  approval_time?: Date;
  status?: ApprovalStatus;
}

export interface Approvals {
  infra?: ApprovalItem;
  risk?: ApprovalItem;
}

export interface Workspace {
  id: number;
  name: string;
  summary: string;
  description: string;
  behavior: string;
  requested_date: Date;
  requester: string;
  single_user: boolean;
  status?: string;
  compliance: Compliance;
  approvals?: Approvals;
  data: HiveAllocation[];
  processing: ResourcePool[];
  topics: KafkaTopic[];
  applications: Application[];
}

export interface UserSuggestion {
  display: string;
  distinguished_name: string;
}

export interface UserSuggestions {
  filter: string;
  users: UserSuggestion[];
  groups: UserSuggestion[];
}

export interface WorkspaceSearchResult {
  id: number;
  name: string;
  summary: string;
  status: string;
  behavior: string;
  pii_data?: boolean;
  pci_data?: boolean;
  phi_data?: boolean;
  date_requested: Date;
  date_fully_approved: Date;
  total_disk_allocated_in_gb: number;
  total_disk_consumed_in_gb?: number;
  total_max_cores: number;
  total_max_memory_in_gb: number;
}
