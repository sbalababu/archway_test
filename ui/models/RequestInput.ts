import { Compliance } from './Workspace';

export interface RequestInput {
  name: string;
  summary: string;
  description: string;
  compliance: Compliance;
}
