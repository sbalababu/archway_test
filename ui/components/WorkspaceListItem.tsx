import * as React from 'react';
import { Card, Icon, List } from 'antd';
import { Doughnut } from 'react-chartjs-2';
import Colors from './Colors';
import { Workspace, WorkspaceSearchResult } from '../models/Workspace';
import TruncateText from './TruncateText';

interface Props {
  workspace: Workspace | WorkspaceSearchResult;
  onSelected: (id: number) => void;
}

const WorkspaceListItem = ({ workspace, onSelected }: Props) => {
  const { id, name, status = '', behavior, summary } = workspace;

  let total_disk_allocated_in_gb;
  let total_disk_consumed_in_gb;
  const workspaceSearchResult = workspace as WorkspaceSearchResult;
  const workspaceData = (workspace as Workspace).data;
  if (workspaceSearchResult.total_disk_allocated_in_gb) {
    total_disk_allocated_in_gb = workspaceSearchResult.total_disk_allocated_in_gb;
    total_disk_consumed_in_gb = workspaceSearchResult.total_disk_consumed_in_gb;
  } else if (workspaceData) {
    total_disk_allocated_in_gb = workspaceData.reduce((acc, cur) => acc + cur.size_in_gb, 0);
    total_disk_consumed_in_gb = workspaceData.reduce((acc, cur) => acc + cur.consumed_in_gb, 0);
  }

  const onClick = () => onSelected(id);

  const ApprovalMessage = () => {
    let color = '';
    switch (status.toLowerCase()) {
      case 'pending':
        color = '#CFB2B0';
        break;
      case 'rejected':
        color = '#7B2D26';
        break;
      case 'approved':
        color = '#0B7A75';
        break;
      default:
    }

    return (
      <div
        style={{
          textTransform: 'uppercase',
          fontSize: 10,
          fontWeight: 300,
          color,
        }}
      >
        {status}
      </div>
    );
  };

  const allocated = total_disk_allocated_in_gb || 1;
  const consumed = total_disk_consumed_in_gb || 0;
  const sizeData = {
    labels: ['Available (GB)', 'Consumed (GB)'],
    datasets: [
      {
        label: false,
        data: [allocated - consumed, consumed],
        backgroundColor: [
          total_disk_consumed_in_gb ? Colors.PrimaryColor.string() : Colors.LightGray.string(),
          total_disk_consumed_in_gb
            ? Colors.PrimaryColor.lighten(0.5).string()
            : Colors.LightGray.lighten(0.5).string(),
        ],
      },
    ],
  };

  return (
    <List.Item>
      <Card bordered={true} onClick={onClick} hoverable={true} bodyStyle={{ padding: '20px' }}>
        <ApprovalMessage />
        <div style={{ display: 'flex', alignItems: 'center', padding: '28px 0 40px' }}>
          <div style={{ fontSize: '12px', color: 'rgba(0, 0, 0, 0.65)', flex: 7 }}>
            <div style={{ fontSize: '22px', textTransform: 'uppercase' }}>
              <TruncateText text={name} lineHeight={32} maxLine={1} />
            </div>
            <div>
              <TruncateText text={summary} lineHeight={16} maxLine={2} />
            </div>
            <div
              style={{
                textTransform: 'uppercase',
                display: 'flex',
                justifyContent: 'flex-start',
                alignItems: 'center',
              }}
            >
              <Icon type={behavior === 'simple' ? 'team' : 'deployment-unit'} />
              &nbsp;
              <TruncateText text={`${behavior} dataset`} lineHeight={12} maxLine={1} />
            </div>
            <div style={{ color: Colors.PrimaryColor.string(), lineHeight: '24px' }}>DETAILS ></div>
          </div>
          <div style={{ flex: 3 }}>
            <div
              style={{
                textAlign: 'center',
                display: 'flex',
                flexDirection: 'column',
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }}
            >
              <Doughnut
                height={52}
                width={52}
                // @ts-ignore
                data={sizeData}
                redraw={false}
                // @ts-ignore
                options={{ legend: false, title: false, maintainAspectRatio: false }}
              />
            </div>
            <div style={{ letterSpacing: 1, textAlign: 'center' }}>
              {`${(allocated - consumed).toFixed(1)}GB/${allocated}GB`}
            </div>
          </div>
        </div>
      </Card>
    </List.Item>
  );
};

export default WorkspaceListItem;
