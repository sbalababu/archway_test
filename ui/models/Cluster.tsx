import { Dropdown, Icon, Menu } from 'antd';
import * as Color from 'color';
import * as React from 'react';
import { Colors } from '../components';

export interface WebLocation {
  host: string;
  port: number;
}

export interface Statusable {
  status: string;
}

export class Status<T extends Statusable> {
  public statusable: T;

  constructor(statusable: T) {
    this.statusable = statusable;
  }

  public statusColor = (): Color => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return Colors.Green;
      case 'CONCERNING_HEALTH':
        return Colors.Orange;
      case 'BAD_HEALTH':
        return Colors.Red;
      default:
        return Colors.Gray;
    }
  };

  public statusText = () => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return 'good';
      case 'CONCERNING_HEALTH':
        return 'concerning';
      case 'BAD_HEALTH':
        return 'bad';
      default:
        return 'unknown';
    }
  };

  public glowColorText = () =>
    `0 0 5px 2px ${this.statusColor()
      .hsl()
      .string()}`;
}

export interface HueService extends Statusable {
  load_balancer: WebLocation[];
}

/* tslint:disable:no-empty-interface */
export interface HiveService extends Statusable {
  thrift: WebLocation[];
}

export interface YarnService extends Statusable {
  resource_manager: WebLocation[];
  node_manager: WebLocation[];
}

export interface NavigatorService extends Statusable {
  navigator: WebLocation[];
}

export abstract class ServiceLinks<T extends Statusable> {
  public service: T;

  public links!: JSX.Element[];

  constructor(service: T) {
    this.service = service;
  }
}

export class HueServiceLinks extends ServiceLinks<HueService> {
  public hueServiceLinks: JSX.Element[] =
    this.service &&
    this.service.load_balancer.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Hue UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.hueServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class HiveServiceLinks extends ServiceLinks<HiveService> {
  public links: JSX.Element[] = [0].map(index => (
    <h4 key={index} style={{ cursor: 'not-allowed', color: 'rgb(0,0,0,.45)', margin: 0 }}>
      (no links)
    </h4>
  ));
}

export class YarnServiceLinks extends ServiceLinks<YarnService> {
  public resourceManagerLinks: JSX.Element[] =
    this.service &&
    this.service.resource_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Resource Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public nodeManagerLinks: JSX.Element[] =
    this.service &&
    this.service.node_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Node Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown
      overlay={
        <Menu>
          {this.resourceManagerLinks}
          {this.nodeManagerLinks}
        </Menu>
      }
    >
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class NavigatorServiceLinks extends ServiceLinks<NavigatorService> {
  public navigatorServiceLinks: JSX.Element[] =
    this.service &&
    this.service.navigator.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Navigator UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.navigatorServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export interface Cluster extends Statusable {
  name: string;
  cm_url: string;
  services: {
    hive: HiveService;
    hue: HueService;
    yarn: YarnService;
    mgmt: NavigatorService;
  };
}
