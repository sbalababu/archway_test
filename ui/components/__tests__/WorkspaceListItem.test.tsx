import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WorkspaceListItem from '../WorkspaceListItem';
import { WorkspaceSearchResult } from '../../models/Workspace';

describe('WorkspaceListItem', () => {
  it('renders correctly', () => {
    const workspace: WorkspaceSearchResult = {
      id: 0,
      name: '',
      summary: '',
      behavior: 'simple',
      status: 'Approved',
      date_requested: new Date(0),
      date_fully_approved: new Date(0),
      total_disk_allocated_in_gb: 0,
      total_max_cores: 0,
      total_max_memory_in_gb: 0,
    };
    const wrapper = shallow(<WorkspaceListItem workspace={workspace} onSelected={() => null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
