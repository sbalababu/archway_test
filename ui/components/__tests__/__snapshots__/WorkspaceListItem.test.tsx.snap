// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceListItem renders correctly 1`] = `
<Item>
  <Card
    bodyStyle={
      Object {
        "padding": "20px",
      }
    }
    bordered={true}
    hoverable={true}
    onClick={[Function]}
  >
    <ApprovalMessage />
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "padding": "28px 0 40px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.65)",
            "flex": 7,
            "fontSize": "12px",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": "22px",
              "textTransform": "uppercase",
            }
          }
        >
          <TruncateText
            lineHeight={32}
            maxLine={1}
            text=""
          />
        </div>
        <div>
          <TruncateText
            lineHeight={16}
            maxLine={2}
            text=""
          />
        </div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "flex-start",
              "textTransform": "uppercase",
            }
          }
        >
          <Icon
            type="team"
          />
          Â 
          <TruncateText
            lineHeight={12}
            maxLine={1}
            text="simple dataset"
          />
        </div>
        <div
          style={
            Object {
              "color": "rgb(45, 116, 147)",
              "lineHeight": "24px",
            }
          }
        >
          DETAILS &gt;
        </div>
      </div>
      <div
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "textAlign": "center",
            }
          }
        >
          <Doughnut
            data={
              Object {
                "datasets": Array [
                  Object {
                    "backgroundColor": Array [
                      "rgb(240, 243, 245)",
                      "hsl(204, 20%, 100%)",
                    ],
                    "data": Array [
                      1,
                      0,
                    ],
                    "label": false,
                  },
                ],
                "labels": Array [
                  "Available (GB)",
                  "Consumed (GB)",
                ],
              }
            }
            height={52}
            options={
              Object {
                "legend": false,
                "maintainAspectRatio": false,
                "title": false,
              }
            }
            redraw={false}
            width={52}
          />
        </div>
        <div
          style={
            Object {
              "letterSpacing": 1,
              "textAlign": "center",
            }
          }
        >
          1.0GB/1GB
        </div>
      </div>
    </div>
  </Card>
</Item>
`;
